aerogear/mobile-security-service-operator
# feat(olm) : Add mobile-security-service-operator.v0.1.0.clusterserviceversion.yaml (AEROGEAR-8872)

## Motivation
https://issues.jboss.org/browse/AEROGEAR-8762

## What
Implementation to allow the project to be available in OLM catalogue and MDC


## Why
Users/developer should be able to use the Operator to create the Mobile Security Service App and Database in their namespaces, however, they should still not able to create the Operator. 

## How
* Using the operator sdk to create the ClusterServiceVersion, following the command used. 

```
operator-sdk  olm-catalog gen-csv --csv-version 0.1.0 --csv-config mobile-security-service-operator.0.1.0.clusterserviceversion.yaml
```

* Creating the other resources which are used by the OLM in the same directory. (CatalogSource, InstallPlan, OperatorGroup, Subscription)
* Creating make commands to deploy and undeploy the olm

## Verification Steps
1. Install OLM in the local Minishift (https://dev4devs.com/2019/03/25/kubernates-openshift-how-to-install-deploy-use-olm-operator-lifecycle-manager-in-minishift/)
2. Update the host as described in the README to config (CR)
3. Run the command `make deploy-olm`
4. Perform the following checks  

[ ] - The namespace `mobile-security-service-operator` should be created
[ ] - The Operator should be installed in the namespace `mobile-security-service-operator`
[ ] - The developer catalogue should show the Mobile Security Service Operator App
[ ] - The developer catalogue should show the Mobile Security Service Operator Database
[ ] - The subscription should appear in the OLM
[ ] - The planInstall should appear in the OLM
[ ] - The OLM should show the operator installed with success
[ ] - Should be possible to create instances with the APP and Database types. 

5. Check that the make undeploy-olm will remove the operator and all related from the cluster. 
 
## Checklist:


## Progress

- [] Finished task
- Fix the error faced which is not giving any clue why the operator is not installed
- Add steps to install OLM and test it with Minishift in the Readme
- Add a section to describe the steps to deploy-undeploy the operator
